lxd/instance/instance_utils.go

```diff
@@ -1075,11 +1085,17 @@ func IsSameLogicalInstance(inst Instance, dbInst *db.InstanceArgs) bool {
 		return true
 	}
 
-	// Instance UUID is expected to be globally unique (which then allows for the *temporary* existence of
-	// duplicate instances of different names with the same volatile.uuid in order to accommodate moving
-	// instances between projects and storage pools without triggering duplicate resource errors).
+	// Don't trigger duplicate resource errors for temporary copies.
 	if dbInst.Config["volatile.uuid"] == inst.LocalConfig()["volatile.uuid"] {
-		return true
+		// Accommodate moving instances between storage pools
+		if IsTemporaryName(dbInst.Name) || IsTemporaryName(inst.Name()) {
```

Comment:
This still isn't comparing the instance names to the specific temporary instance name that we would consider for skipping. Its just parsing the uuid now and checking for validity, but not matching to the one we expect.

Reply:
Ah ok, got it. Right now it just checks in general if its a temp name and accepts it. Has the same effect, since it only does this check if both UUIDs are identical, but the check itself is "too broad".

Reply:
Or lets call it not intuitive :)

Reply:
Yes exactly